playwright/all_tests/assignment/assignment.spec.ts

```diff
@@ -245,3 +245,70 @@ test.describe('Presentation view', () => {
     await testKatseluNakyma(page, `assignment-${assignment.id}-action-katselunakyma`, assignmentIn)
   })
 })
+
+const favoritesCount = async (page: Page): Promise<string> => {
+  await page.waitForTimeout(1000)
+  return await page.getByTestId('header-favorites-count').innerText()
+}
```

Comment:
Kun päivittää favorite statuksen tapahtuu erittäin nopea state päivitys, mutta ei tarpeeksi nopea, koska playwright ehtii ottaa vanhan arvon. Timeoutilla saa menemää läpi, mutta haluttais kyllä tästä eroon.

Reply:
Vois kuvitella, että playwright tarjoais tavan retryttää ja odottaa, että jokin ehto toteutuu.

Reply:
Ehkä tää expect.poll: https://playwright.dev/docs/test-assertions#expectpoll

Reply:
Tää oli mahtava löyty 👍 kiitooos